### PR TITLE
Support for 3-way validation split interface

### LINKED ISSUE #697: Support `initial_validation_split` objects
dev rsample gained a new interface for validation splits, `initial_validation_split()` and friends, which produce objects of class `initial_validation_split`.  The follow-up function `validation_set()` produces an `rset` like the old interface `validation_split()`. (https://github.com/tidymodels/rsample/pull/403 and https://github.com/tidymodels/rsample/pull/404)

tune should know what to do with these objects. I think this essentially means we need to adapt `last_fit()` because it takes a spilt object as input rather than an rest.

``` r
library(tidymodels) # with dev rsample

val_split <- initial_validation_split(mtcars) # new class
val_set <- validation_set(val_split) # returns the same class as the old interface

spline_rec <- recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp)
lin_mod <- linear_reg() %>%
  set_engine("lm")

# fit_resamples() and tune_*() work with rsets
spline_res <- fit_resamples(lin_mod, spline_rec, resamples = val_set,
                            control = control_resamples(save_pred = TRUE))
#> Warning in `[.tbl_df`(x, is.finite(x <- as.numeric(x))): NAs introduced by
#> coercion
try_augment_without_printing <- augment(spline_res)
#> Warning: The orginal data had 25 rows but there were 6 hold-out predictions.


spline_rec_tune <- recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp, deg_free = tune())

spline_res <- tune_grid(lin_mod, spline_rec_tune, resamples = val_set, 
                        control = control_grid(save_workflow = TRUE, save_pred = TRUE))
#> Warning in `[.tbl_df`(x, is.finite(x <- as.numeric(x))): NAs introduced by
#> coercion
#> → A | warning: prediction from a rank-deficient fit may be misleading
#> There were issues with some computations   A: x1
#> There were issues with some computations   A: x3
#> 

try_fit_best_without_printing <- fit_best(spline_res)
try_augment_without_printing <- augment(spline_res)
#> Warning: The orginal data had 25 rows but there were 6 hold-out predictions.


# this one needs adaption
spline_res <- last_fit(lin_mod, spline_rec, split = val_split)
#> Error in `new_rset()`:
#> ! Each element of `splits` must be an `rsplit` object.
#> Backtrace:
#>     ▆
#>  1. ├─tune::last_fit(lin_mod, spline_rec, split = val_split)
#>  2. └─tune:::last_fit.model_spec(lin_mod, spline_rec, split = val_split)
#>  3.   └─tune:::last_fit_workflow(wflow, split, metrics, control)
#>  4.     └─rsample::manual_rset(splits, ids = "train/test split")
#>  5.       └─rsample:::new_manual_rset(splits, ids)
#>  6.         └─rsample::new_rset(splits, ids, subclass = c("manual_rset", "rset"))
#>  7.           └─rlang::abort("Each element of `splits` must be an `rsplit` object.")
#spline_res
#augment(spline_res)
```

<sup>Created on 2023-06-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by hfrick**
closes #697

This PR adds support for the new interface to validation sets in rsample via `initial_validation_split()` and `validation_set()`. 

As laid out in https://github.com/tidymodels/tune/issues/697#issuecomment-1620267038, this 

- extends `last_fit()` to work with objects coming from `initial_validation_split()` 
- adds an argument `use_validation_set` to `last_fit()` and `fit_best()` 
- sets defaults for `use_validation_set` so that we keep using the validation set for objects coming from the old interface but do not use it for objects from the new interface 

To make that possible, rsample adds another attribute to validation `rset` objects coming from the new interface `validation_set()` in https://github.com/tidymodels/rsample/pull/437

For reference, we will deprecate the old interface: https://github.com/tidymodels/rsample/issues/350

### CODE REVIEW

File: R/fit_best.R [hunk 1]
```diff
@@ -84,6 +90,7 @@ fit_best.tune_results <- function(x,
                                   metric = NULL,
                                   parameters = NULL,
                                   verbose = FALSE,
+                                  use_validation_set = NULL,
```
Comment (topepo) [inline R/fit_best.R:93]: Let's call this `train_with_validation` instead?

File: R/last_fit.R [hunk 2]
```diff
@@ -166,6 +175,32 @@ last_fit_workflow <- function(object,
     )
   }
 
+
+  if (inherits(split, "initial_validation_split")) {
```
Comment (topepo) [inline R/last_fit.R:179]: This block should go into a different function to make the main function more readable.

**Review by topepo — COMMENTED** [review]
One comment but otherwise great. I'll wait for the name changes from rsample to do anything with this.

File: R/last_fit.R (hunk 2 above)
Comment (hfrick) [reply R/last_fit.R:178]: adressed in [842a85a](https://github.com/tidymodels/tune/pull/701/commits/842a85a8241c24b84bdc5acc608bb30aac549aeb)

File: R/fit_best.R (hunk 1 above)
Comment (hfrick) [reply R/fit_best.R]: after the team discussion we went with `add_validation_set`

**Pull request merged**